File: seleniumEasyPython/steps/simple_form_steps.py
from behave import Given, When, Then
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
from selenium.webdriver.support import expected_conditions


@Given(u'the developer is on the demo selenium easy website')
def get_selenium_easy_website(context):
    context.driver.get("https://demo.seleniumeasy.com/basic-first-form-demo.html")


@When(u'the developer closes the pop-up advertisement')
def close_pop_up_advertisement(context):
    context.simple_input.pop_up_window().click()


#     ------------------- Single Input Fields --------------------------------


@When(u'the developer enters a message into the input box')
def enter_message_in_input(context):
    context.simple_input.message_input().send_keys("I entered this message with Selenium.")


@When(u'the developer clicks on the show message button')
def click_show_message_button(context):
    context.simple_input.show_message_button().click()


@Then(u'the message typed in the input box will display on the page')
def message_displayed(context):
    element = WebDriverWait(context.driver, 4).until(
        expected_conditions.text_to_be_present_in_element((By.ID, "display"),
                                                          "I entered this message with Selenium."))
    # assert element.text == "I entered this message with Selenium."
    assert element


# ------------------- Two Input Fields --------------------------------
@When(u'the developer enters a value into the a input box')
def enter_first_value(context):
    context.simple_input.input_a().send_keys("2")


@When(u'the developer enters a value into the b input box')
def enter_second_value(context):
    context.simple_input.input_b().send_keys("2")


@When(u'the developer clicks on the total button')
def click_total_button(context):
    context.simple_input.show_value_button().click()


@Then(u'the total will display on the page')
def total_displayed(context):
    element = WebDriverWait(context.driver, 4).until(
        expected_conditions.text_to_be_present_in_element((By.ID, "displayvalue"),
                                                          "4"))
    # assert element.text == "4"
    assert element

File: seleniumEasyJava/src/test/java/E2E/steps/simpleInput/SimpleFormSteps.java
package E2E.steps.simpleInput;

import E2E.runner.TestRunner;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.junit.Assert;

public class SimpleFormSteps {
    // ------------------- Single Input Field --------------------------------

    @Given("the developer is on the demo selenium easy website")
    public void theDeveloperIsOnTheDemoSeleniumEasyWebsite() {
        TestRunner.driver.get("https://demo.seleniumeasy.com/basic-first-form-demo.html");
    }

    @When("the developer closes the pop-up advertisement")
    public void theDeveloperClosesThePopUpAdvertisement() {
        TestRunner.simpleInputPage.popUpWindowCloseAnchorTag.click();
    }

    @When("the developer enters a message into the input box")
    public void theDeveloperEntersAMessageIntoTheInputBox() {
        TestRunner.simpleInputPage.messageInput.sendKeys("I entered this message with Selenium.");

    }

    @When("the developer clicks on the show message button")
    public void theDeveloperClicksOnTheShowMessageButton() {
        TestRunner.simpleInputPage.showMessageButton.click();

    }

    @Then("the message typed in the input box will display on the page")
    public void theMessageTypedInTheInputBoxWillDisplayOnThePage() {
        TestRunner.explicitWait.until(ExpectedConditions.textToBePresentInElement(TestRunner.simpleInputPage
                .displayMessageSpan, "I entered this message with Selenium."));
        String message = TestRunner.simpleInputPage.displayMessageSpan.getText();
        Assert.assertEquals(message, "I entered this message with Selenium.");
    }

    // ------------------- Two Input Fields --------------------------------

    @When("the developer enters a value into the a input box")
    public void theDeveloperEntersAValueIntoTheAInputBox() {
        TestRunner.simpleInputPage.inputA.sendKeys("1");
    }

    @When("the developer enters a value into the b input box")
    public void theDeveloperEntersAValueIntoTheBInputBox() {
        TestRunner.simpleInputPage.inputB.sendKeys("1");
    }

    @When("the developer clicks on the total button")
    public void theDeveloperClicksOnTheTotalButton() {
        TestRunner.simpleInputPage.getTotalButton.click();
    }

    @Then("the total will display on the page")
    public void theTotalWillDisplayOnThePage() {
        TestRunner.explicitWait.until(ExpectedConditions.textToBePresentInElement(TestRunner.simpleInputPage
                .displayValueSpan, "2"));
        String message = TestRunner.simpleInputPage.displayValueSpan.getText();
        Assert.assertEquals(message, "2");
    }
}
